message view: Do not mark selected message as read on narrow.

ryanreh99 opened the pull request:
I have another branch where I'm working on the last read message selection approach.

Now, the selected message isn't read for the narrows in the left sidebar.
The entire message list gets marked as read if the bottom message is visible
on the narrow activation, this behavior remains the same .
Also works for the All Messages view.

WIP as tests are remaining and I'll do some more manual testing.
Fixes #10886.

showell commented on static/js/message_scroll.js:
```diff
@@ -166,7 +166,7 @@ exports.initialize = function () {
     // Scroll handler that marks messages as read when you scroll past them.
     $(document).on("message_selected.zulip", (event) => {
-        if (event.id === -1) {
+        if (event.id === -1 || event.id === event.previously_selected) {
```
I wonder if a good follow up fix here would be to rename `event.id` and `event.previously_selected`.  It's not obvious that either is a message id, and when you look at this code out of context, it seems funny to compare those two things.

timabbott commented:
I merged the first commit, which is a nice refactor; we might also want to rename `event.id` to `event.now_selected_id` for better readability.

ryanreh99 commented:
I rebased this and reworded the commit message and removed `Fixes #10886` from it.

The commit message explains why the `event.id === event.previously_selected_id` logic works.

Now to begin marking messages as read, the user has to interact with the middle message feed.
Except for the case where all the messages fit within the viewport.


Also, one point to note is that due to the above condition above it seems that the following flow interaction occurs:

* You narrow to a page
* The blue pointer is on the first unread message
* You click the selected message, but the message doesn't get marked as read due to the above condition

However the reselected message does gets marked as read due to this [event handler](https://github.com/zulip/zulip/blob/master/static/js/click_handlers.js#L181).

timabbott commented:
So, thinking about this PR, I had been envisioning a different approach for fixing #10886, where we'd:
* When narrowing, select the message just before the first unread message, rather than the first unread message.
* Select the area at the top of the page with the Zulip logo, if all messages are unread.

I think that's a more intuitive model, since then we preserve the property that selecting a message marks it as read, rather than it depending how you ended up selecting the message.  Thoughts?

bew commented:
Is this still relevant now that #10886 has been [closed](https://github.com/zulip/zulip/issues/10886#issuecomment-2229075459) ?
